src/librustc/session/config.rs

```diff
@@ -223,6 +223,15 @@ impl OutputFilenames {
     pub fn filestem(&self) -> String {
         format!("{}{}", self.out_filestem, self.extra)
     }
+
+    pub fn is_path_used(&self, path: PathBuf) -> bool {
+        for flavor in self.outputs.keys() {
+            if self.path(*flavor) == path {
```

Comment:
I felt that doing it this way I was hedging against `path` ever changing behaviour. Having said that, I just pushed a change that doesn't use `path` for this check.